holoviews/plotting/bokeh/callbacks.py

```diff
@@ -663,15 +665,14 @@ def _update_selection_event(self, event):
         self._selection_event = event
         self._processed_event = not event.final
         if event.final and self._skipped_partial_event:
-            self._process_selection_event()
-            self._skipped_partial_event = False
+            async_execute(self._process_selection_partial_event)
 
     def on_msg(self, msg):
         super().on_msg(msg)
         if hasattr(self, '_panel'):
-            self._process_selection_event()
+            async_execute(self._process_selection_event)
```

Comment:
Not convinced `async_execute` is correct here.

Reply:
Making all the relevant methods async doesn't seem to work unless there's a Bokeh upstream change if I'm not mistaken:
```
[/Users/ahuang/miniconda3/envs/lumen/lib/python3.10/site-packages/bokeh/util/callback_manager.py:107](https://file+.vscode-resource.vscode-cdn.net/Users/ahuang/miniconda3/envs/lumen/lib/python3.10/site-packages/bokeh/util/callback_manager.py:107): RuntimeWarning: coroutine 'PopupMixin._update_selection_event' was never awaited
  cast(EventCallbackWithEvent, callback)(event)
```

```
    async def _update_selection_event(self, event):
        if (((prev:= self._selection_event) and prev.final and not self._processed_event) or
            self.geom_type not in (event.geometry["type"], "any")):
            return
        self._selection_event = event
        self._processed_event = not event.final
        if event.final and self._skipped_partial_event:
            await self._process_selection_partial_event()

    async def on_msg(self, msg):
        await super().on_msg(msg)
        if hasattr(self, '_panel'):
            await self._process_selection_event()
```